### PR TITLE
U4-11492 Clean Umbraco.Web.UmbracoHelper [v8]

### PR DESCRIPTION
**Pull request by Mundairson**
### Prerequisites

- [X] I have written a descriptive pull-request title
- [X] I have linked this PR to an issue on the tracker at http://issues.umbraco.org/issue/U4-11492

### Description
Refactored the v8 versions of the below methods.

```
class Umbraco.Web.UmbracoHelper

bool ConvertIdObjectToGuid(object id, out Guid guidId)
bool ConvertIdObjectToUdi(object id, out Udi guidId)
bool ConvertIdObjectToInt(object id, out int intId)
```

### CODE REVIEW

File: src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs [hunk 1]
```diff
@@ -1,83 +1,90 @@
...
+                "FormAction",
+                "",
+                additionalRouteValues
+                );
+
             var result = encryptedRouteString.DecryptWithMachineKey();
-            var expectedResult = "c=FormController&a=FormAction&ar=&key1=value1&key2=value2&Key3=Value3&keY4=valuE4";
 
-            Assert.AreEqual(expectedResult, result);
+            const string EXPECTED_RESULT = "c=FormController&a=FormAction&ar=&key1=value1&key2=value2&Key3=Value3&keY4=valuE4";
```
Comment (nul800sebastiaan) [inline src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs]: Variable name convention for constants are the same as for normal vars, so we'd use `expectedResult`.

File: src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs [hunk 2]
```diff
@@ -86,29 +93,35 @@ public void Create_Encrypted_RouteString_From_Dictionary()
...
```
Comment (nul800sebastiaan) [inline src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs]: Variable name should be `expectedResult`.

File: src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs [hunk 3]
```diff
@@ -118,21 +131,19 @@ public void Truncate_By_Words_With_Tag()
...
 
             Assert.AreEqual("Hello world, this is some text <a href='blah'>with&hellip;</a>", result);
         }
 
         [Test]
-        public void Strip_All_Html()
+        public static void Strip_All_Html()
         {
-            var text = "Hello world, <b>this</b> is some text <a href='blah'>with a link</a>";
+            const string text = "Hello world, <b>this</b> is some text <a href='blah'>with a link</a>";
```
Comment (nul800sebastiaan) [inline src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs]: This string with the `<b>` tag in it repeats a few times, should it also be a field, like `Sample`?

File: src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs [hunk 4]
```diff
@@ -156,15 +167,256 @@ public void Strip_Specific_Html()
...
             var helper = new UmbracoHelper();
 
             var result = helper.StripHtml(text).ToString();
 
             Assert.AreEqual("Hello world, is some text with a link", result);
         }
+
+        // ------- Int32 conversion tests
+        [Test]
+        public static void Converting_boxed_34_to_an_int_returns_34()
```
Comment (nul800sebastiaan) [inline src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs]: Make sure to capitalize each letter after an underscore.

File: src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs [hunk 5]
```diff
@@ -156,15 +167,256 @@ public void Strip_Specific_Html()
...
+                out int result
+                );
+
+            // Assert
+            Assert.IsTrue(success);
+            Assert.That(result, Is.EqualTo(34));
+        }
+
+        [Test]
+        public static void Converting_string_54_to_an_int_returns_54()
```
Comment (nul800sebastiaan) [inline src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs]: Make sure to capitalize each letter after an underscore.

File: src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs [hunk 6]
```diff
@@ -156,15 +167,256 @@ public void Strip_Specific_Html()
...
+                out int result
+                );
+
+            // Assert
+            Assert.IsTrue(success);
+            Assert.That(result, Is.EqualTo(54));
+        }
+
+        [Test]
+        public static void Converting_hello_to_an_int_returns_false()
```
Comment (nul800sebastiaan) [inline src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs]: Make sure to capitalize each letter after an underscore.

File: src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs [hunk 7]
```diff
@@ -156,15 +167,256 @@ public void Strip_Specific_Html()
...
+                out int result
+                );
+
+            // Assert
+            Assert.IsFalse(success);
+            Assert.That(result, Is.EqualTo(0));
+        }
+
+        [Test]
+        public static void Converting_unsupported_object_to_an_int_returns_false()
```
Comment (nul800sebastiaan) [inline src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs]: Make sure to capitalize each letter after an underscore.

File: src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs [hunk 8]
```diff
@@ -156,15 +167,256 @@ public void Strip_Specific_Html()
...
+                );
+
+            // Assert
+            Assert.IsFalse(success);
+            Assert.That(result, Is.EqualTo(0));
+        }
+
+        // ------- GUID conversion tests
+        [Test]
+        public static void Converting_boxed_guid_to_a_guid_returns_original_guid_value()
```
Comment (nul800sebastiaan) [inline src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs]: Make sure to capitalize each letter after an underscore.

File: src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs [hunk 9]
```diff
@@ -156,15 +167,256 @@ public void Strip_Specific_Html()
...
+                out Guid result
+                );
+
+            // Assert
+            Assert.IsTrue(success);
+            Assert.That(result, Is.EqualTo(sample));
+        }
+
+        [Test]
+        public static void Converting_string_guid_to_a_guid_returns_original_guid_value()
```
Comment (nul800sebastiaan) [inline src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs]: Make sure to capitalize each letter after an underscore.

File: src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs [hunk 10]
```diff
@@ -156,15 +167,256 @@ public void Strip_Specific_Html()
...
+                out Guid result
+                );
+
+            // Assert
+            Assert.IsFalse(success);
+            Assert.That(result, Is.EqualTo(new Guid("00000000-0000-0000-0000-000000000000")));
+        }
+
+        [Test]
+        public static void Converting_unsupported_object_to_a_guid_returns_false()
```
Comment (nul800sebastiaan) [inline src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs]: Make sure to capitalize each letter after an underscore.

File: src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs [hunk 11]
```diff
@@ -156,15 +167,256 @@ public void Strip_Specific_Html()
...
+                );
+
+            // Assert
+            Assert.IsFalse(success);
+            Assert.That(result, Is.EqualTo(new Guid("00000000-0000-0000-0000-000000000000")));
+        }
+
+        // ------- UDI Conversion Tests
+        [Test]
+        public void Converting_boxed_udi_to_a_udi_returns_original_udi_value()
```
Comment (nul800sebastiaan) [inline src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs]: Make sure to capitalize each letter after an underscore.

File: src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs [hunk 12]
```diff
@@ -156,15 +167,256 @@ public void Strip_Specific_Html()
...
+
+            // Assert
+            TearDownUdiTests();
+
+            Assert.IsTrue(success);
+            Assert.That(result, Is.EqualTo(sample));
+        }
+
+        [Test]
+        public void Converting_string_udi_to_a_udi_returns_original_udi_value()
```
Comment (nul800sebastiaan) [inline src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs]: Make sure to capitalize each letter after an underscore.

File: src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs [hunk 13]
```diff
@@ -156,15 +167,256 @@ public void Strip_Specific_Html()
...
+
+            // Assert
+            TearDownUdiTests();
+
+            Assert.IsTrue(success, "Conversion of UDI failed.");
+            Assert.That(result, Is.EqualTo(sample));
+        }
+
+        [Test]
+        public void Converting_hello_to_a_udi_returns_false()
```
Comment (nul800sebastiaan) [inline src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs]: Make sure to capitalize each letter after an underscore.

File: src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs [hunk 14]
```diff
@@ -156,15 +167,256 @@ public void Strip_Specific_Html()
...
+
+            // Assert
+            TearDownUdiTests();
+
+            Assert.IsFalse(success);
+            Assert.That(result, Is.Null);
+        }
+
+        [Test]
+        public void Converting_unsupported_object_to_a_udi_returns_false()
```
Comment (nul800sebastiaan) [inline src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs]: Make sure to capitalize each letter after an underscore.

File: src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs [hunk 15]
```diff
@@ -156,15 +167,256 @@ public void Strip_Specific_Html()
...
+                );
+
+            // Assert
+            TearDownUdiTests();
+
+            Assert.IsFalse(success);
+            Assert.That(result, Is.Null);
+        }
+
+        public void SetUpUdiTests()
```
Comment (nul800sebastiaan) [inline src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs]: Make sure to rename this method to make clear what it does.

File: src/Umbraco.Web/UmbracoHelper.cs [hunk 16]
```diff
@@ -601,37 +597,40 @@ public IEnumerable<IPublishedContent> ContentAtRoot()
...
+                default:
+                    intId = default;
+                    return false;
             }
-            intId = default(int);
-            return false;
         }
 
-        private static bool ConvertIdObjectToGuid(object id, out Guid guidId)
+        /// <remarks>Had to change to internal for testing.</remarks>
```
Comment (nul800sebastiaan) [inline src/Umbraco.Web/UmbracoHelper.cs]: No need for this comment to be in source code, just leave a comment in Github on the PR next time.

File: src/Umbraco.Web/UmbracoHelper.cs [hunk 17]
```diff
@@ -665,17 +664,22 @@ private static bool ConvertIdsObjectToGuids(IEnumerable<object> ids, out IEnumer
             return true;
         }
 
-        private static bool ConvertIdObjectToUdi(object id, out Udi guidId)
+        /// <remarks>Had to change to internal for testing.</remarks>
```
Comment (nul800sebastiaan) [inline src/Umbraco.Web/UmbracoHelper.cs]: No need for this comment to be in source code.

**Comment by Mundairson** [comment]
Hi @nul800sebastiaan 

Thanks for the detailed feedback! I'll make the changes as requested but, for my future reference, the code is the mostly the same as I submitted for the v7 branch so are the v7 and v8 coding standards substantially different? None of this was flagged for that PR.

**Comment by nul800sebastiaan** [comment]
@Mundairson  🙈 Should've caught that then as well!
I must have been extra-enthusiastic to merge at the time! 😉

**Comment by Mundairson** [comment]
@nul800sebastiaan Haha ok, no problem. I'll make the same reviewed changes for the v7 branch and submit a PR for those.

Is there a list of coding standards somewhere? The `.editorconfig` file is quite empty and (unless I've completely missed it) I can't find a standards file anywhere. Indeed, should the `.editorconfig` file be updated to help developers?

Thanks!

**Comment by nul800sebastiaan** [comment]
@Mundairson I've resisted writing up our "coding standards" as it stops people from contributing. If they get prescribed exactly what to do, they get anxious beforehand and don't send in their contribution because they're scared they missed something.

It should be pretty easy, `.editorconfig` prescribes a few little things, then just follow existing coding styles and almost always follow ReSharper's hints (in case you have it installed).

I'd much rather give contributors personal feedback and add an emoji, it's the Umbraco Friendly way! 👍

**Comment by nul800sebastiaan** [comment]
Looks great now, thanks very much!

**Pull request merged**